examples/07_advanced/plot_mask_large_fmri.py

```diff
@@ -0,0 +1,279 @@
+"""
+Working with large fMRI images
+==============================
+
+In this example, we will demonstrate how one can work with large fMRI images
+more efficiently.
+
+Particularly, we will consider a case where we have a large fMRI image
+and we want to extract the data from several regions of interest (ROIs) defined
+by a number of binary masks, all in parallel.
+
+The issue when trying to process a large fMRI image in parallel like this is
+that each parallel process will load the entire fMRI image into memory. This
+can lead to a significant increase in memory usage and can slow down the
+processing.
+
+We will compare three different methods to mask the data from the fMRI image:
+
+1. Using the :class:`~nilearn.maskers.NiftiMasker`
+2. Using numpy indexing
+3. Using numpy indexing with shared memory
+
+"""
+
+# %%
+# Create a large fMRI image
+# -------------------------
+# Here we will create a "large" fMRI image by fetch 5 subjects'
+# fMRI images via the :func:`~nilearn.datasets.fetch_development_fmri`
+# function, concatenating them and then saving to a file.
+
+from pathlib import Path
+
+from nilearn.datasets import fetch_development_fmri
+from nilearn.image import concat_imgs
+
+N_SUBJECTS = 5
+N_REGIONS = 6
+
+fmri_data = fetch_development_fmri(n_subjects=N_SUBJECTS)
+fmri_img = concat_imgs(fmri_data.func)
```

Comment:
Do we want to do this with real data ? It doesn't seem to make the example _too_ heavy, although I remember that was a concern

Reply:
Yeah I wanted to keep it reasonable for the doc so that it doesn't slow it down even further.

But I made sure to check much heavier versions on an HPC side-by-side (like in this [comment](https://github.com/nilearn/nilearn/pull/5113#issuecomment-2627902466)) and the results seem to align -- at least between the HPC and the doc build.